core/src/main/scala/org/apache/spark/storage/StorageUtils.scala

```diff
@@ -35,7 +35,13 @@ import org.apache.spark.internal.Logging
  * class cannot mutate the source of the information. Accesses are not thread-safe.
  */
 @DeveloperApi
-class StorageStatus(val blockManagerId: BlockManagerId, val maxMem: Long) {
+class StorageStatus(
+    val blockManagerId: BlockManagerId,
+    // Explicitly adding this maxMemory field to handle maxOnHeapMem and maxOffHeapMem not
+    // existing issue, this happened when trying to replay old event log.
+    val maxMemory: Long,
+    val maxOnHeapMem: Option[Long],
+    val maxOffHeapMem: Option[Long]) {
```

Comment:
I find the comment you have here a little confusing.  I'd get rid of it and put in something more like what you have where you create the `StorageStatus` in `BlockManagerMasterEndpoint`, something like

The onHeap and offHeap memory are always defined for new applications, but they can be missing if we are replaying old event logs.